Mathlib/Analysis/SpecialFunctions/Trigonometric/Cotangent.lean

```diff
@@ -223,3 +232,178 @@ theorem cot_series_rep (hz : x ∈ ℂ_ℤ) :
+  obtain ⟨A, B, hB, HABK⟩ := subset_verticalStrip_of_isCompact
+    ((isCompact_iff_isCompact_univ.mp hKc).image_of_continuousOn
+    (continuous_inclusion hK |>.continuousOn))
+  exact ⟨cotTermUpperBound k A B hB, summable_cotTermUpperBound A B hB hk,
+    iteratedDerivWithin_cotTerm_bounded_uniformly hk hK A B hB HABK⟩
+
+theorem differentiableOn_iteratedDerivWithin_cotTerm (n l : ℕ) :
+    DifferentiableOn ℂ (iteratedDerivWithin l (fun z ↦ cotTerm z n) ℍₒ) ℍₒ := by
+  suffices DifferentiableOn ℂ (fun z : ℂ ↦ (-1) ^ l * l ! * ((z + (n + 1)) ^ (-1 - l : ℤ) +
+    (z - (n + 1)) ^ (-1 - l : ℤ))) ℍₒ by
+    exact this.congr fun z hz ↦ eqOn_upperHalfPlaneSet_iteratedDerivWithin_cotTerm l n hz
+  apply DifferentiableOn.const_mul
+  apply DifferentiableOn.add <;> apply DifferentiableOn.zpow
+  any_goals try {fun_prop} <;> left <;> intro x hx
+  · simpa [add_eq_zero_iff_neg_eq'] using (UpperHalfPlane.ne_int ⟨x, hx⟩ (-(n + 1))).symm
+  · simpa [sub_eq_zero] using (UpperHalfPlane.ne_int ⟨x, hx⟩ (n + 1))
+
+private theorem aux_summable_add {k : ℕ} (hk : 1 ≤ k) (x : ℂ) :
+  Summable fun (n : ℕ) ↦ (x + (n + 1)) ^ (-1 - k : ℤ) := by
+  apply ((summable_nat_add_iff 1).mpr (summable_int_iff_summable_nat_and_neg.mp
+        (EisensteinSeries.linear_right_summable x 1 (k := k + 1) (by omega))).1).congr
+  simp [← zpow_neg, sub_eq_add_neg]
+
+private theorem aux_summable_neg {k : ℕ} (hk : 1 ≤ k) (x : ℂ) :
+  Summable fun (n : ℕ) ↦ (x - (n + 1)) ^ (-1 - k : ℤ) := by
+  apply ((summable_nat_add_iff 1).mpr (summable_int_iff_summable_nat_and_neg.mp
+        (EisensteinSeries.linear_right_summable x 1 (k := k + 1) (by omega))).2).congr
+  simp [← zpow_neg, sub_eq_add_neg]
+
+variable {z : ℂ}
+
+-- We have this auxiliary ugly version on the lhs so the the rhs looks nicer.
+private theorem aux_iteratedDeriv_tsum_cotTerm {k : ℕ} (hk : 1 ≤ k) (hz : z ∈ ℍₒ) :
+    (-1) ^ k * (k !) * z ^ (-1 - k : ℤ) + iteratedDerivWithin k
+    (fun z ↦ ∑' n : ℕ, cotTerm z n) ℍₒ z =
+    (-1) ^ k * k ! * ∑' n : ℤ, (z + n) ^ (-1 - k : ℤ) := by
+  rw [iteratedDerivWithin_tsum k isOpen_upperHalfPlaneSet hz
+    (fun t ht ↦ Summable_cotTerm (coe_mem_integerComplement ⟨t, ht⟩))
+    (fun l hl hl2 ↦ summableLocallyUniformlyOn_iteratedDerivWithin_cotTerm  hl)
+    (fun n l z hl hz ↦ (differentiableOn_iteratedDerivWithin_cotTerm n l).differentiableAt
+    (isOpen_upperHalfPlaneSet.mem_nhds hz))]
+  conv =>
+    enter [1,2,1]
+    ext n
+    rw [eqOn_upperHalfPlaneSet_iteratedDerivWithin_cotTerm k n (by simp [hz])]
+  rw [tsum_of_add_one_of_neg_add_one (by simpa using aux_summable_add hk z)
+    (by simpa [sub_eq_add_neg] using aux_summable_neg hk z),
+    tsum_mul_left, Summable.tsum_add (aux_summable_add hk z) (aux_summable_neg hk z)]
+  simp only [Int.reduceNeg, sub_eq_add_neg, neg_add_rev, Int.cast_add, Int.cast_natCast,
+    Int.cast_one, Int.cast_zero, add_zero, Int.cast_neg]
+  ring
+
+theorem iteratedDerivWithin_cot_sub_inv_eq_series_rep {k : ℕ} (hk : 1 ≤ k) (hz : z ∈ ℍₒ) :
```

Comment:
```suggestion
theorem iteratedDerivWithin_cot_sub_inv_eq_add_mul_tsum {k : ℕ} (hk : 1 ≤ k) (hz : z ∈ ℍₒ) :
```
or perhaps simply `iteratedDerivWithin_cot_sub_inv_eq_tsum`, even though not strictly correct.